pkg/plugins/backendplugin/grpcplugin/grpc_plugin.go

```diff
@@ -134,94 +144,111 @@ func (p *grpcPlugin) Target() backendplugin.Target {
 	return backendplugin.TargetLocal
 }
 
-func (p *grpcPlugin) getPluginClient() (pluginClient, bool) {
+func (p *grpcPlugin) getPluginClient(ctx context.Context) (*ClientV2, bool) {
 	p.mutex.RLock()
-	if p.client == nil || p.client.Exited() || p.pluginClient == nil {
+	if p.client != nil && !p.client.Exited() && p.pluginClient != nil {
 		p.mutex.RUnlock()
-		return nil, false
+		return p.pluginClient, true
```

Comment:
Race problem here reading plugin client after read lock released. Put the client in a variable and then unlock as we did earlier

Reply:
Realized I can just `defer` here instead